packages/openai/src/openai-chat-options.ts

```diff
@@ -92,15 +92,21 @@ export const openaiProviderOptions = z.object({
    * Parameters for prediction mode.
    */
   prediction: z.record(z.any()).optional(),
-});
-
-export type OpenAIProviderOptions = z.infer<typeof openaiProviderOptions>;
 
-export interface OpenAIChatSettings {
   /**
-Whether to use structured outputs. Defaults to false.
+   * Whether to use structured outputs.
+   *
+   * @default false
+   */
+  structuredOutputs: z
+    .union([
+      z.boolean(),
+      z.object({
+        tools: z.boolean().optional(),
+        response: z.boolean().optional(),
+      }),
```

Comment:
Where is the requirement to separate structured outputs for `tools` and `response` coming from? I don't think users asked for this.